bentoml/cli/azure_function.py

```diff
@@ -0,0 +1,423 @@
+        max_burst,
+        premium_plan_sku,
+        labels,
+        function_auth_level,
+        output,
+        wait,
+    ):
+        track_cli('deploy-create', PLATFORM_NAME)
+        bento_name, bento_version = bento.split(':')
+        yatai_client = YataiClient()
+        try:
+            with Spinner('Deploying Azure function deployment'):
+                result = yatai_client.deployment.create_azure_function_deployment(
+                    name=name,
+                    namespace=namespace,
+                    labels=labels,
+                    bento_name=bento_name,
+                    bento_version=bento_version,
+                    location=location,
+                    min_instances=min_instances,
+                    max_burst=max_burst,
+                    premium_plan_sku=premium_plan_sku,
+                    function_auth_level=function_auth_level,
+                    wait=wait,
+                )
+                if result.status.status_code != status_pb2.Status.OK:
+                    error_code, error_message = status_pb_to_error_code_and_message(
+                        result.status
+                    )
+                    _echo(
+                        f'Failed to create Azure function deployment {name} '
+                        f'{error_code}:{error_message}',
+                        CLI_COLOR_ERROR,
+                    )
+                    return
+                track_cli('deploy-create-success', PLATFORM_NAME)
+                _echo(
+                    f'Successfully created Azure function deployment {name}',
+                    CLI_COLOR_SUCCESS,
+                )
+                _print_deployment_info(result.deployment, output)
+        except BentoMLException as e:
+            _echo(
+                f'Failed to create Azure function deployment {name}. {str(e)}',
+                CLI_COLOR_ERROR,
+            )
+
+    @azure_function.command(help='Update existing Azure function deployment')
+    @click.argument('name', type=click.STRING)
+    @click.option(
+        '-b',
+        '--bento',
+        '--bento-service-bundle',
+        type=click.STRING,
+        # callback=parse_bento_tag_callback,
+        help='Target BentoService to be deployed, referenced by its name and version '
+        'in the format of name:version. For example: "iris_classifier:v1.2.0"',
+    )
+    @click.option(
+        '-n',
+        '--namespace',
+        type=click.STRING,
+        help='Deployment namespace managed by BentoML, the default value is "dev" '
+        'which can be changed in BentoML configuration file',
+    )
+    @click.option(
+        '--min-instances',
+        type=click.INT,
+        help='The minimum number of workers for the deployment.',
+    )
+    @click.option(
+        '--max-burst',
+        type=click.INT,
+        help='The maximum number of elastic workers for the deployment.',
+    )
+    @click.option(
+        '--premium-plan-sku',
+        type=click.STRING,
```

Comment:
`ALL_AZURE_PREMIUM_SKU = ['EP1', 'EP2', ...]`
`type=click.Choice(ALL_AZURE_PREMIUM_SKU)`

Reply:
same for function_auth_level